lib/scenic/adapters/postgres/index_reapplication.rb

```diff
@@ -0,0 +1,63 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Dropping a materialized view causes all indexes on the materialized view
+      # to be dropped. This object can be used to capture the existing
+      # materialized views before updating the materialized view and then
+      # reapplying after.
+      #
+      # @api private
+      class IndexReapplication
+        def initialize(connection:, speaker: ActiveRecord::Migration)
+          @connection = connection
+          @speaker = speaker
+        end
+
+        # Caches indexes on the provided object before executing the block and
+        # then reapplying the indexes. Errors in applying the indexes are caught
+        # and logged to output.
```

Comment:
No they're not.

"Errors reapplying the indices cause the index not to be re-created. Each recreated or skipped index will be noted to STDOUT"